packages/components/pagination/src/page-navigator/page-navigator.js

```diff
@@ -14,77 +15,69 @@ import messages from './messages';
-
   const [pageNumberInputId] = React.useState(uniqueId('page-number-'));
 
-  const { onPageChange, totalPages } = props;
+  const paginationForm = useFormik({
+    initialValues: { page: props.page },
+    enableReinitialize: true,
+    validateOnBlur: true,
+    validateOnChange: true,
+    validateOnMount: true,
+    onSubmit: (values /**, helpers */) => {
+      const nextNormalizedValue = Number(
+        normalizePageValue(values.page, props.totalPages)
+      );
+      props.onPageChange(nextNormalizedValue);
+    },
+    validate: (values) => {
+      if (!isValid(values.page, props.totalPages)) {
+        return { page: true };
+      }
+      return {};
+    },
+  });
 
-  const normalizedValue = Number(normalizePageValue(page, totalPages));
+  const { page, totalPages } = props;
   const isDisabled = totalPages === 0;
   const isPreviousDisabled = page <= 1;
   const isNextDisabled = page >= totalPages;
 
-  const handleSubmit = React.useCallback(
-    (event) => {
-      event.preventDefault();
-
-      setPage(normalizedValue);
-
-      onPageChange(normalizedValue);
-    },
-    [normalizedValue, onPageChange]
-  );
+  const handlePrevPageChange = React.useCallback(() => {
+    const previousPage = paginationForm.values.page - 1;
+    if (previousPage < 1) return;
+    paginationForm.setFieldValue('page', previousPage, true);
+    paginationForm.submitForm();
+  }, [paginationForm]);
 
-  const onBlurNormalize = React.useCallback(() => {
-    setPage(normalizedValue);
-  }, [normalizedValue]);
-
-  const handleFocus = React.useCallback((event) => {
-    event.target.select();
-  }, []);
-
-  const handleChange = React.useCallback((event) => {
-    setPage(event.target.value);
-  }, []);
-
-  const handlePrevPage = React.useCallback(() => {
-    const prevPage = page - 1;
-    if (prevPage < 1) return null;
-
-    setPage(prevPage);
-    onPageChange(prevPage);
-  }, [page, onPageChange]);
-
-  const handleNextPage = React.useCallback(() => {
-    const nextPage = page + 1;
+  const handleNextPageChange = React.useCallback(() => {
+    const nextPage = paginationForm.values.page + 1;
     if (nextPage > totalPages) return null;
-
-    setPage(nextPage);
-    onPageChange(nextPage);
-  }, [page, onPageChange, totalPages]);
+    paginationForm.setFieldValue('page', nextPage, true);
+    paginationForm.submitForm();
+  }, [paginationForm, totalPages]);
 
   return (
-    <form onSubmit={handleSubmit}>
+    <form onSubmit={paginationForm.handleSubmit}>
       <Spacings.Inline alignItems="center" scale="s">
         <SecondaryIconButton
           label={intl.formatMessage(messages.previousPageLabel)}
-          onClick={handlePrevPage}
+          onClick={handlePrevPageChange}
           isDisabled={isPreviousDisabled || isDisabled}
           icon={<AngleLeftIcon />}
         />
         <Label htmlFor={pageNumberInputId} intlMessage={messages.page} />
         <div>
           <NumberInput
+            name="page"
```

Comment:
```suggestion
            name="page-input"
```

Reply:
this name should reflect the name of the formik state ;)

Reply:
I know, I just wanted to suggest a more specific name 😉